src/components/grid.js

```diff
@@ -3,36 +3,46 @@ import PropTypes from 'prop-types';
 
 import Tile from './tile';
 
-const Grid = ({ grid, readOnly, onClick }) => (
-    <div className="puzzle-column">
-        {grid.map((row, rowKey) => {
-            return (
-                <div className="puzzle-row" key={rowKey}>
-                    {row.map(
-                        tileValue =>
-                            tileValue === 0 ? (
-                                <div
-                                    key={tileValue}
-                                    className="puzzle-tile-empty"
-                                />
-                            ) : (
-                                <Tile
-                                    key={tileValue}
-                                    enabled={!readOnly}
-                                    onClick={onClick}
-                                    tileValue={tileValue}
-                                />
-                            ),
-                    )}
-                </div>
-            );
-        })}
-    </div>
-);
+import { associateTileToBackground } from '../core/helper';
+
+const Grid = ({ grid, onClick, resolvedGrid, readOnly }) => {
+    const url = '/static/images/sea.jpg';
```

Comment:
If its a constant, you can move this constant out of the Grid.
This way, you won't instanciate `url` in memory for each Grid.